thing/index.js

```diff
@@ -515,7 +526,20 @@ function ThingShadowsClient(deviceOptions, thingShadowOptions) {
          //
          if (thingShadows[thingName].persistentSubscribe === true) {
             this._handleSubscriptions(thingName, ['update', 'get', 'delete'], ['accepted', 'rejected'],
-               'subscribe');
+               'subscribe', function () {
```

Comment:
OK. We should also clean up the subscriptions and remove the shadow. Good to add more error handling.